gapic/schema/wrappers.py

```diff
@@ -593,6 +593,63 @@ def client_output(self):
         # Return the usual output.
         return self.output
 
+    @utils.cached_property
+    def client_output_async(self):
+        """Return the output from the client layer.
```

Comment:
Good idea! I will create a function to hide common logic. In next pass, I will try to improve readability using similar pattern.

Reply:
Thanks. If it turns out to be tricky then no need to twist the logic too much.

Reply:
I didn't find a way to unify this specific method without regression.
1. The `client_output` is cached, so it can't accept parameters;
2. The output of `client_output` is frozen, so we can't modify the content in it (we can create a new one but it is worse than what we have now).

Reply:
Keep `client_output` and `client_output_async` as the cached layer, then move the guts of the logic into an internal method that takes a parameter. The internal method should _not_ be cached and should not be a property.

Reply:
Created the common internal method in 4ecf34dd1e0303210521d7f315a03eb2d5324015.